primitives/types/metadata_generator.go

```diff
@@ -0,0 +1,179 @@
+package types
+
+import (
+	"reflect"
+	"strings"
+
+	sc "github.com/LimeChain/goscale"
+	"github.com/LimeChain/gosemble/constants/metadata"
+)
+
+type MetadataGenerator interface {
```

Comment:
is this interface actually needed? Unlike common OOP languages, interfaces in Go should be avoided unless there's a real use case in the codebase. This would also eliminate the name collision between the interface and the struct. Here's a doc with more examples that are considered best practices for writing idiomatic Go interfaces: [https://github.com/pthethanh/effective-go?tab=readme-ov-file#interfaces](url)

Reply:
Agreed. My initial goal was to make this MetadataGenerator as a util and I was using the other util (MemoryTranslator) as a reference. Due to circular dependencies, I failed making it as a util at the moment.